safedelete/admin.py

```diff
@@ -165,3 +167,80 @@ def undelete_selected(self, request, queryset):
                 context,
             )
     undelete_selected.short_description = _("Undelete selected %(verbose_name_plural)s.")
+
+    def hard_delete(self, request, queryset):
+        """ Admin action to hard delete objects. """
+        if not self.has_delete_permission(request):
+            raise PermissionDenied
+
+        original_queryset = queryset.all()
+        undeleted_queryset = queryset.filter(deleted__isnull=True)
+        queryset = queryset.filter(deleted__isnull=False)
+
+        if request.POST.get('post'):
+            requested = original_queryset.count()
+            changed = queryset.count()
+
+            if changed:
+                for obj in queryset:
+                    obj.delete(force_policy=HARD_DELETE)
```

Comment:
It would be better to call the base QuerySet delete to delete in bulk, otherwise it could be very long.